controllers/kubeadmconfig_controller.go

```diff
@@ -449,30 +455,40 @@ func (r *KubeadmConfigReconciler) createClusterCertificates(ctx context.Context,
 		return nil, err
 	}
 
-	secret := &corev1.Secret{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      ClusterCertificatesSecretName(clusterName),
-			Namespace: config.GetNamespace(),
-			OwnerReferences: []v1.OwnerReference{
-				{
-					APIVersion: cabpkv1alpha2.GroupVersion.String(),
-					Kind:       "KubeadmConfig",
-					Name:       config.GetName(),
-					UID:        config.GetUID(),
-				},
-			},
-		},
-		Data: certificates.ToMap(),
+	for _, secret := range certs.NewSecretsFromCertificates(certificates) {
+		linkOwnerReference(&secret.ObjectMeta, config)
+		secret.ObjectMeta.Labels["cluster"] = clusterName
```

Comment:
Let's use the same label we already have - `"cluster.x-k8s.io/cluster-name"`. We have a constant for this, `MachineClusterLabelName`, in `sigs.k8s.io/cluster-api/api/v1alpha2`. We can either use this, or create our own constant in this repo.